### PR TITLE
Nuke IO Layer

### PR DESCRIPTION
**Pull request by akhilveeraghanta**
### Description
* Remove IO layer and all fw stuff that is no longer supported
* Cleanup CI and remove Firmware Job
* Cleanup setup scripts (get rid of CubeMX, openocd, fw setup, etc..)
* Move platformio setup to setup_software
* Move estop_arduino_uno to `software/emergency_stop`

### Testing Done
Builds

### Resolved Issues
N/A

#### :warning:  A trick would be to filter out `.h, .c, .ld and .S` files, (estop stuff has a .cc extension so it won't get filtered)

Please _only_ review:

* setup_software.sh
* software/emergency_stop/*
* .github/main.yaml

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

**Comment by jonathanlew** [comment]
it resolves https://github.com/UBC-Thunderbots/Software/issues/1475

File: src/.bazelrc [hunk 1]
```diff
@@ -10,8 +10,6 @@ test --test_output=all
 build --crosstool_top=//cc_toolchain:toolchain
 
 # Add warnings to Thunderbots code only
-build --per_file_copt=//software/.*,//shared/.*,-//software/proto/*,//firmware/.*@-Wall,-Wextra,-Wconversion,-Wno-unused-parameter,-Werror
-
```
Comment (jonathanlew) [inline src/.bazelrc]: we need this

**Review by jonathanlew — CHANGES_REQUESTED** [review]
nice

**Review by amrm3lm — DISMISSED** [review]
end of an era

File: src/.bazelrc [hunk 2]
```diff
@@ -10,6 +10,8 @@ test --test_output=all
 build --crosstool_top=//cc_toolchain:toolchain
 
 # Add warnings to Thunderbots code only
+build --per_file_copt=//software/.*,//shared/.*,-//software/proto/*,//firmware/.*@-Wall,-Wextra,-Wconversion,-Wno-unused-parameter,-Werror
```
Comment (jonathanlew) [inline src/.bazelrc]:
```suggestion
build --per_file_copt=//software/.*,//shared/.*@-Wall,-Wextra,-Wconversion,-Wno-unused-parameter,-Werror
```

**Review by jonathanlew — COMMENTED** [review]
just a bit of clean up left

File: src/.bazelrc (hunk 1 above)
Comment (akhilveeraghanta) [reply src/.bazelrc]: LOL good catch

**Review by jonathanlew — APPROVED** [review]
🧹

**Review by amrm3lm — APPROVED** [review]

**Pull request merged**